src/shogun/kernel/CombinedKernel.cpp

```diff
@@ -66,27 +66,20 @@ bool CCombinedKernel::init_with_extracted_subsets(
     SGVector<index_t> rhs_subset)
 {
 
+	auto l_combined = dynamic_cast<CCombinedFeatures*>(l);
+	auto r_combined = dynamic_cast<CCombinedFeatures*>(r);
+
+	if (!l_combined || !r_combined)
+		SG_ERROR("Cast failed - unsupported features passed")
```

Comment:
imo this will never happen (unless l or r is null) since if the typecasting fails it'll throw a bad_casting exception.